data_safe_haven/commands/users.py

```diff
@@ -146,15 +146,27 @@ def register(
 
         # List users
         users = UserHandler(context, graph_api)
-        available_usernames = users.get_usernames_entra_id()
+        available_users = users.entra_users.list()
```

Comment:
I think the name `available_users` is a bit confusing here. Especially when we are checking if the username is already taken.

Reply:
Actually, no, this is the register command, right?

Reply:
This is the register command, yes. I chose `available_users` because the output is a list of `ResearchUsers` - not just usernames, but the full records. 
We need the user principal name here, as we need to see what domain they're attached to.

`get_usernames_entra_id()` just gets the first part of the user principal name.

It's possible we may want a `get_user_principal_names` function, as we may want to do the same thing for the `add` function - but I'll look at doing that when it comes to working on that.